### PR TITLE
MES-1506: added transform method arn functionality, and completed unit test coverage

### PR DESCRIPTION
**Pull request by robshakespeare**
- Due to the way API gateway caches authorization policies, the ARN must allow access to all resources within the application stage.  This PR includes a commit which changes the httpVerb and resourcePath parts of an API gateway Amazon Resource Name (ARN) in to * wildcards.  E.g. with an input methodArn of `arn:aws:execute-api:region:account:app/stage/GET/journal/10023488/personal` the result policy methodArn will be `arn:aws:execute-api:region:account:app/stage/*/*`

- This PR also brings the unit test coverage to 100%

### CODE REVIEW

File: src/functions/authoriser/application/__tests__/transformMethodArn.spec.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    expect(() => sut(''))
+      .toThrow(new Error('Failed to extract arnBase and stageName from methodArn: ""'));
+  });
+
+  it('throws error if passed whitespace string', () => {
+    expect(() => sut(' \t '))
+      .toThrow(new Error('Failed to extract arnBase and stageName from methodArn: " \t "'));
+  });
+
+  it('throws error if passed null string', () => {
```
Comment (johnmorleybjss) [inline src/functions/authoriser/application/__tests__/transformMethodArn.spec.ts]: Do we have strictness set high enough that the compiler wouldn't allow spurious nulls/undefined?

Comment (robshakespeare) [reply src/functions/authoriser/application/__tests__/transformMethodArn.spec.ts]: @johnmorleybjss Its a good point, yes the strictness would pick up nulls passed in the Typescript.  The problem here is that the `methodArn` originates from outside of this service, the `methodArn` is passed by the AWS infrastructure.  Although I expect AWS or its configuration never to change to an extent that it cause a null to be passed, what I wanted here was a simple check to ensure we didn't get horrible to debug null reference errors.

**Comment by robshakespeare** [comment]
> Looks really good.

Thanks

**Comment by robshakespeare** [comment]
> Great stuff, the more I look at typemoq, the more I think we should probably be using it elsewhere.

Good to hear, thanks @johnmorleybjss

File: src/functions/authoriser/application/__tests__/transformMethodArn.spec.ts (hunk 1 above)
Comment (johnmorleybjss) [reply src/functions/authoriser/application/__tests__/transformMethodArn.spec.ts]: I see that the type of `CustomAuthorizerEvent.methodArn` is `string`, not `string | null | undefined`, so with the strictness levels as they are, there's no way a null/undefined can sneak through at compile time. However, banking on this at runtime seems like asking for trouble. Happy to leave it as is for safety's sake. 👍

**Pull request merged**